diff --git a/tests/invalid_models_tests/test_relative_fields.py b/tests/invalid_models_tests/test_relative_fields.py
index a6edcc4..24409e9 100644
--- a/tests/invalid_models_tests/test_relative_fields.py
+++ b/tests/invalid_models_tests/test_relative_fields.py
@@ -2500,6 +2500,48 @@ class DatabaseLevelOnDeleteTests(TestCase):
             ],
         )
 
+    @skipUnlessDBFeature("supports_on_delete_db_cascade")
+    def test_db_python_m2m_chain(self):
+        class OtherModelParent(models.Model):
+            pass
+
+        class OtherModel(models.Model):
+            parent = models.ForeignKey(OtherModelParent, on_delete=models.DB_CASCADE)
+
+        class Parent(models.Model):
+            pass
+
+        class Child(models.Model):
+            parent = models.ForeignKey(Parent, on_delete=models.DB_CASCADE)
+            other_models = models.ManyToManyField(OtherModel)
+
+        field = Child._meta.get_field("other_models")
+        self.assertEqual(
+            field.check(from_model=Child, databases=self.databases),
+            [
+                Error(
+                    "Field specifies database-level on_delete variant, but "
+                    "auto-created intermediary model uses Python-level variant.",
+                    hint=(
+                        "Use either one of the Python on_delete variants or create a "
+                        "through model for invalid_models_tests.Child.other_models."
+                    ),
+                    obj=Child._meta.get_field("parent"),
+                    id="fields.E323",
+                ),
+                Error(
+                    "Field specifies database-level on_delete variant, but "
+                    "auto-created intermediary model uses Python-level variant.",
+                    hint=(
+                        "Use either one of the Python on_delete variants or create a "
+                        "through model for invalid_models_tests.Child.other_models."
+                    ),
+                    obj=OtherModel._meta.get_field("parent"),
+                    id="fields.E323",
+                ),
+            ],
+        )
+
     @skipUnlessDBFeature("supports_on_delete_db_cascade")
     def test_db_python_chain_auto_created(self):
         class GrandParent(models.Model):
