diff --git a/tests/migrations/test_autodetector.py b/tests/migrations/test_autodetector.py
index f2bc7e4..7fc6f6b 100644
--- a/tests/migrations/test_autodetector.py
+++ b/tests/migrations/test_autodetector.py
@@ -1,6 +1,7 @@
 import copy
 import functools
 import re
+import uuid
 from unittest import mock
 
 from django.apps import apps
@@ -757,6 +758,16 @@ class AutodetectorTests(BaseAutodetectorTests):
         {"managed": False},
         ("testapp.author",),
     )
+    author_unmanaged_uid_unique_default = ModelState(
+        "testapp",
+        "Author",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("uid", models.UUIDField(default=uuid.uuid4, unique=True)),
+        ],
+        {"managed": False},
+        ("testapp.author",),
+    )
     author_unmanaged_name_longer = ModelState(
         "testapp",
         "Author",
@@ -767,6 +778,26 @@ class AutodetectorTests(BaseAutodetectorTests):
         {"managed": False},
         ("testapp.author",),
     )
+    author_unmanaged_name_no_default = ModelState(
+        "testapp",
+        "Author",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("name", models.CharField(max_length=200)),
+        ],
+        {"managed": False},
+        ("testapp.author",),
+    )
+    author_unmanaged_name_nullable = ModelState(
+        "testapp",
+        "Author",
+        [
+            ("id", models.AutoField(primary_key=True)),
+            ("name", models.CharField(max_length=200, null=True)),
+        ],
+        {"managed": False},
+        ("testapp.author",),
+    )
     author_unmanaged_renamed_with_book = ModelState(
         "testapp",
         "Writer",
@@ -4132,6 +4163,14 @@ class AutodetectorTests(BaseAutodetectorTests):
         self.assertOperationTypes(changes, "testapp", 0, ["AddField"])
         self.assertOperationAttributes(changes, "testapp", 0, 0, name="name")
 
+    def test_unmanaged_add_field_unique_default(self):
+        changes = self.get_changes(
+            [self.author_unmanaged_empty], [self.author_unmanaged_uid_unique_default]
+        )
+        self.assertNumberMigrations(changes, "testapp", 1)
+        self.assertOperationTypes(changes, "testapp", 0, ["AddField"])
+        self.assertOperationAttributes(changes, "testapp", 0, 0, name="uid")
+
     def test_unmanaged_alter_field(self):
         """Tests autodetection of altered fields on an unmanaged model."""
         changes = self.get_changes(
@@ -4144,6 +4183,17 @@ class AutodetectorTests(BaseAutodetectorTests):
             changes, "testapp", 0, 0, name="name", preserve_default=True
         )
 
+    def test_unmanaged_alter_field_no_default(self):
+        changes = self.get_changes(
+            [self.author_unmanaged_name_nullable],
+            [self.author_unmanaged_name_no_default],
+        )
+        self.assertNumberMigrations(changes, "testapp", 1)
+        self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])
+        self.assertOperationAttributes(
+            changes, "testapp", 0, 0, name="name", preserve_default=True
+        )
+
     def test_unmanaged_create(self):
         """The autodetector correctly deals with unmanaged models."""
         # First, we test adding an unmanaged model
