diff --git a/tests/migrations/test_operations.py b/tests/migrations/test_operations.py
index 939520d..9bbf124 100644
--- a/tests/migrations/test_operations.py
+++ b/tests/migrations/test_operations.py
@@ -2573,6 +2573,33 @@ class OperationTests(OperationTestBase):
                 operation.database_forwards(app_label, editor, new_state, project_state)
         self.assertColumnExists(rider_table, "pony_id")
 
+    def test_alter_field_python_level_on_delete_noop(self):
+        """
+        AlterField operation is a noop when changing only the Python-level
+        on_delete option.
+        """
+        app_label = "test_alflodnoop"
+        project_state = self.set_up_test_model(app_label, related_model=True)
+        operation = migrations.AlterField(
+            "Rider", "pony", models.ForeignKey("Pony", models.PROTECT)
+        )
+        new_state = project_state.clone()
+        operation.state_forwards(app_label, new_state)
+        self.assertIs(
+            project_state.models[app_label, "rider"]
+            .fields["pony"]
+            .remote_field.on_delete,
+            models.CASCADE,
+        )
+        self.assertIs(
+            new_state.models[app_label, "rider"].fields["pony"].remote_field.on_delete,
+            models.PROTECT,
+        )
+        with connection.schema_editor() as editor, self.assertNumQueries(0):
+            operation.database_forwards(app_label, editor, project_state, new_state)
+        with connection.schema_editor() as editor, self.assertNumQueries(0):
+            operation.database_backwards(app_label, editor, new_state, project_state)
+
     def test_alter_field_foreignobject_noop(self):
         app_label = "test_alflfo_noop"
         project_state = self.set_up_test_model(app_label)
diff --git a/tests/schema/tests.py b/tests/schema/tests.py
index 024e68f..292df44 100644
--- a/tests/schema/tests.py
+++ b/tests/schema/tests.py
@@ -635,10 +635,57 @@ class SchemaTests(TransactionTestCase):
         ):
             editor.alter_field(Book, old_field, new_field)
         self.assertForeignKeyExists(Book, "author_id", "schema_author")
+        self.assertGreater(len(ctx.captured_queries), 0)
         self.assertIs(
             any("ON DELETE" in query["sql"] for query in ctx.captured_queries), False
         )
 
+    @skipUnlessDBFeature("supports_foreign_keys", "can_introspect_foreign_keys")
+    def test_fk_alter_on_delete_python_level_noop(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+        old_field = Book._meta.get_field("author")
+        new_field = ForeignKey(Author, PROTECT)
+        new_field.set_attributes_from_name("author")
+        # Changing between Python-level on_delete options doesn't require
+        # database changes.
+        with connection.schema_editor() as editor, self.assertNumQueries(0):
+            editor.alter_field(Book, old_field, new_field, strict=True)
+        with connection.schema_editor() as editor, self.assertNumQueries(0):
+            editor.alter_field(Book, new_field, old_field, strict=True)
+
+    @isolate_apps("schema")
+    @skipUnlessDBFeature("supports_foreign_keys", "can_introspect_foreign_keys")
+    def test_fk_alter_on_delete_db_level(self):
+        class DBOnDeleteParent(Model):
+            class Meta:
+                app_label = "schema"
+
+        class DBOnDeleteChild(Model):
+            parent = ForeignKey(DBOnDeleteParent, DB_CASCADE, null=True)
+
+            class Meta:
+                app_label = "schema"
+
+        self.isolated_local_models = [DBOnDeleteChild, DBOnDeleteParent]
+        with connection.schema_editor() as editor:
+            editor.create_model(DBOnDeleteParent)
+            editor.create_model(DBOnDeleteChild)
+        # Changing between database-level on_delete options requires database
+        # changes.
+        old_field = DBOnDeleteChild._meta.get_field("parent")
+        new_field = ForeignKey(DBOnDeleteParent, DB_SET_NULL, null=True)
+        new_field.set_attributes_from_name("parent")
+        with (
+            connection.schema_editor() as editor,
+            CaptureQueriesContext(connection) as ctx,
+        ):
+            editor.alter_field(DBOnDeleteChild, old_field, new_field, strict=True)
+        self.assertIs(
+            any("SET NULL" in query["sql"] for query in ctx.captured_queries), True
+        )
+
     @isolate_apps("schema")
     @skipUnlessDBFeature("supports_foreign_keys", "can_introspect_foreign_keys")
     def test_create_model_db_on_delete(self):
@@ -4926,7 +4973,7 @@ class SchemaTests(TransactionTestCase):
             error_messages={"invalid": "error message"},
             help_text="help text",
             limit_choices_to={"limit": "choice"},
-            on_delete=CASCADE,
+            on_delete=PROTECT,
             related_name="related_name",
             related_query_name="related_query_name",
             validators=[lambda x: x],
